In [1]:
from sklearn import datasets
iris = datasets.load_iris() #load dataser iris
X = iris.data  # we only take the first two features.
y = iris.target

In [3]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, train_size = 0.7)

In [4]:
print("Banyak data latih setelah dilakukan Train-Validation Split: ", len(X_train))
print("Banyak data uji setelah dilakukan Train-Validation Split: ", len(X_test))

Banyak data latih setelah dilakukan Train-Validation Split:  105
Banyak data uji setelah dilakukan Train-Validation Split:  45


In [5]:
from sklearn.model_selection import cross_val_score
from sklearn.svm import SVC

model = SVC(kernel = 'linear', C = 1)
scores = cross_val_score(model, X, y, cv = 5)
print("Akurasi model SVM untuk tiap fold: ", scores)
print("Akurasi model SVM dengan 5-Fold Cross Validation: ", scores.mean())

Akurasi model SVM untuk tiap fold:  [0.96666667 1.         0.96666667 0.96666667 1.        ]
Akurasi model SVM dengan 5-Fold Cross Validation:  0.9800000000000001


In [6]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn import metrics

knn = KNeighborsClassifier()

knn.fit(X_train, y_train)
y_pred = knn.predict(X_test)
score = metrics.accuracy_score(y_test, y_pred)
print("Akurasi dengan menggunakan Nearest Neighbor: ", score)

Akurasi dengan menggunakan Nearest Neighbor:  0.9555555555555556


In [7]:
# Import kNN Classifier dari sklearn
from sklearn.neighbors import KNeighborsClassifier

kkn = KNeighborsClassifier(n_neighbors=3) # memilih 3 sebagai banyaknya neighbors

In [8]:
from sklearn.tree import DecisionTreeClassifier
from sklearn import metrics

dt = DecisionTreeClassifier(
  max_depth = None,
  min_samples_split = 2
)

dt.fit(X_train, y_train)
y_pred = dt.predict(X_test)
score = metrics.accuracy_score(y_test, y_pred)
print("Akurasi dengan menggunakan Decision Tree: ", score)

Akurasi dengan menggunakan Decision Tree:  0.9555555555555556


In [9]:
# Import Decisiom Tree Classifier dari sklearn 
from sklearn.tree import DecisionTreeClassifier

dt = DecisionTreeClassifier(
        max_depth=4, # menentukan kedalaman maksimum tree = 4
        min_samples_split = 2 # menentukan 2 minimal sampel agar node bisa split
)

In [10]:
from sklearn import naive_bayes
from sklearn import metrics

nb = naive_bayes.BernoulliNB()

nb.fit(X_train, y_train)
y_pred = nb.predict(X_test)
score = metrics.accuracy_score(y_test, y_pred)
print("Akurasi dengan menggunakan Naive Bayes: ", score)

Akurasi dengan menggunakan Naive Bayes:  0.28888888888888886


In [11]:
# Import Suppotrt Vector Machine Classifier dari sklearn
from sklearn.svm import SVC

svmliniear = SVC(
        kernel = "linear",
        C = 1
      )

svmPoly = SVC(
        kernel = "poly",
        C = 1,
        gamma = 0.01,
        degree = 2
      )

svmRBF = SVC(
        kernel = "rbf",
        C = 1,
        gamma = 0.01
      )

In [12]:
from sklearn.svm import SVC
from sklearn import metrics

svm = SVC(
  kernel = 'rbf',
  C = 1,
  gamma = 0.01
)

svm.fit(X_train, y_train)
y_pred = svm.predict(X_test)
score = metrics.accuracy_score(y_test, y_pred)
print("Akurasi dengan menggunakan Support Vector Machine: ", score)

Akurasi dengan menggunakan Support Vector Machine:  0.8888888888888888


In [14]:
# Import AdaBoost Classifier dari sklearn
from sklearn.ensemble import AdaBoostClassifier

# Import metric untuk memeriksa akurasi
from sklearn import metrics

ab = AdaBoostClassifier(
        n_estimators = 50,
        learning_rate = 1
      )

ab.fit(X_train, y_train)
y_pred = ab.predict(X_test)
score = metrics.accuracy_score(y_test, y_pred)
print(score)

0.9555555555555556


In [15]:
# Import AdaBoost Classifier dari sklearn
from sklearn.ensemble import AdaBoostClassifier

ab = AdaBoostClassifier(
        n_estimators = 24, # menentukan banyaknya weak learner sebanyak 24
        learning_rate = 1 # menentukan learning rate sebesar 1
)

Dari Hasil Analisis Perbandingan Terhadap Performa Yang Dihasilkan dalam hal akurasi dapat disimpulkan bahwa Metode Klasifikasi K-Nearset Neighbors, Decision Tree dan AdaBoost Classifier memiliki akurasi yang paling tinggi yaitu dengan score 0.955555555555555, sedangkan Metode Klasifikasi SVM memiliki score akurasi 0.8888888888888888 lebih rendah dari ketiga metode diatas namun memiliki socre lebih tinggi dari metode Naive Bayes dengan score akurasi 0.28888888888888886. Jadi dapat disimpulkan urutan akurasi yang tertinggi dan terendah adalah K-Nearest Neighbors, Decision Tree, dan AdaBoost Classifeir, lalu dibawahnya ada SVM (Support Vector Machine) dan yang paling rendah score akurasinya itu ialah Metode Naive Bayes.